sklearn/compose/tests/test_column_transformer.py

```diff
@@ -1397,7 +1397,7 @@ def test_make_column_selector_with_select_dtypes(cols, pattern, include, exclude
         {
             "col_int": np.array([0, 1, 2], dtype=int),
             "col_float": np.array([0.0, 1.0, 2.0], dtype=float),
-            "col_str": ["one", "two", "three"],
+            "col_str": pd.Series(["one", "two", "three"], dtype=object),
```

Comment:
I'd rather update the parametrisation, i.e. fix the `include` param to handle strings as well, to not promote the old pandas behavior in the test suite.

Reply:
Updated to do that (it makes the diff a bit larger, and in the end this is mostly testing that the parameters are correctly passed through to pandas, so personally I think keeping as object would also be fine)

Reply:
I was expecting that we can leave this line unchanged, or even that we should leave it unchanged so that we test both the old and new behaviour depending on the pandas version.

Like this we only check the (somewhat weird?) case where a user makes a string column but explicitly sets the dtype to `object`.

Reply:
Ah yes, my suggestion was to update the parametrization of the test instead of updating this line.